nautobot/extras/datasources/git.py

```diff
@@ -908,7 +911,49 @@ def delete_git_config_context_schemas(repository_record, job_result, preserve=()
 
 def refresh_git_jobs(repository_record, job_result, delete=False):
     """Callback function for GitRepository updates - refresh all Job records managed by this repository."""
-    # No-op as jobs are not currently stored in the DB but are instead refreshed on-request.
+    installed_jobs = []
+    if not delete:
+        jobs_path = os.path.join(repository_record.filesystem_path, "jobs")
+        if os.path.isdir(jobs_path):
+            for job_info in jobs_in_directory(jobs_path):
+                job_model, created = refresh_job_model_from_job_class(
+                    Job,
+                    f"{JobSourceChoices.SOURCE_GIT}.{repository_record.slug}",
+                    job_info.job_class,
+                )
+                if created:
+                    job_result.log(
+                        f'Created Job model "{job_model}"',
+                        grouping="jobs",
+                        level_choice=LogLevelChoices.LOG_SUCCESS,
+                        logger=logger,
+                    )
+                else:
+                    job_result.log(
+                        f'Refreshed Job model "{job_model}"',
+                        grouping="jobs",
+                        level_choice=LogLevelChoices.LOG_SUCCESS,
+                        logger=logger,
+                    )
```

Comment:
On these lines the only thing that differs is the message. Could we just parameterize that one and drop a few lines?
```suggestion
                if created:
                    message = f'Created Job model "{job_model}"'
                else:
                    messge = f'Refreshed Job model "{job_model}"'
                job_result.log(
                    message=message,
                    grouping="jobs",
                    level_choice=LogLevelChoices.LOG_SUCCESS,
                    logger=logger,
                )     
```